compiler/rustc_error_codes/src/error_codes/E0311.md

```diff
@@ -0,0 +1,44 @@
+This error occurs when there is an unsatisfied outlives bound involving an
+elided region on a generic type parameter or associated type.
+
+Erroneous code example:
+
+```compile_fail,E0311
+fn no_restriction<T>(x: &()) -> &() {
+    with_restriction::<T>(x)
+}
+
+fn with_restriction<'a, T: 'a>(x: &'a ()) -> &'a () {
+    x
+}
+```
+
+Why doesn't this code compile? It helps to look at the lifetime bounds that are
+automatically added by the compiler. For more details see the documentation for
+[lifetime elision]( https://doc.rust-lang.org/reference/lifetime-elision.html).
+
+The compiler elides the lifetime of `x` and the return type to some arbitrary
+lifetime `'anon` in `no_restriction()`. The only information available to the
+compiler is that `'anon` is valid for the duration of the function. When
+calling `with_restriction()`, the compiler requires the completely unrelated
+type parameter `T` to outlive `'anon` because of the `T: 'a` bound in
+`with_restriction()`. This causes an error because `T` is not required to
+outlive `'anon` in `no_restriction()`.
+
+If `no_restriction()` were to use `&T` instead of `&()` as an argument, the
+compiler would have added an implied bound [implied
```

Comment:
```suggestion
compiler would have added an [implied
```

(Remove the double "implied bound")

Alternatively, remove the link for now, and then send a new PR when the reference PR lands. I don't think it's worth blocking on the reference PR.

Reply:
I had figured that updating the docs would be quick, but it looks like it's going to be a more involved process.  For now I removed the implied bounds link and will plan to merge this PR. Once https://github.com/rust-lang/reference/pull/1261 has been merged I'll plan to open a smaller PR to add the reference link into the `E0311.md` docs.